### PR TITLE
Add dataset metadata for new E&A to all relevant datasets

### LINKED ISSUE #303: [E & A] Add dataset markdown and layer structured data to all GHG datasets for E&A
The new E&A interface requires some additional layer information in particular structures, such that it can be surfaced in the interface.

In the longer run, we aim at storing and fetching this information from STAC, but for until then, a bit of information duplication is unfortunately required.

Please see details on which data to provide and its form in 

- https://github.com/NASA-IMPACT/veda-ui/pull/852

## Acceptance criteria

- [x] https://github.com/NASA-IMPACT/veda-ui/issues/831
- [ ] https://github.com/US-GHG-Center/veda-config-ghg/pull/317

### PR DESCRIPTION
**Pull request by j08lue**
Closes #303 

1. Add `infoDescription` blocks for each dataset
2. Add `info` blocks for each layer
3. Add `media` blocks for each layer

# Structured info (info description)
- [x] casagfed-carbonflux-monthgrid-v3
- [x] eccodarwin-co2flux-monthgrid-v5
- [x] gosat-based-ch4budget-yeargrid-v1
- [x] lpjwsl-wetlandch4-grid-v1
- [x] oco2-mip-co2budget-yeargrid-v1
- [x] oco2geos-co2-daygrid-v10r
- [x] odiac-ffco2-monthgrid-v2022
- [x] sedac-popdensity-yeargrid5yr-v4.11
- [x] tm54dvar-ch4flux-monthgrid-v1
- [x] epa-ch4emission-yeargrid-v2express
- [x] noaa-cpfp-ch4-point
- [x] noaa-cpfp-co2-point
- [x] emit-ch4plume-v1

# Structured info (info blocks for each layer)
- [x] casagfed-carbonflux-monthgrid-v3
- [x] eccodarwin-co2flux-monthgrid-v5
- [x] gosat-based-ch4budget-yeargrid-v1
- [x] lpjwsl-wetlandch4-grid-v1
- [x] oco2-mip-co2budget-yeargrid-v1
- [x] oco2geos-co2-daygrid-v10r
- [x] odiac-ffco2-monthgrid-v2022
- [x] sedac-popdensity-yeargrid5yr-v4.11
- [x] tm54dvar-ch4flux-monthgrid-v1
- [x] epa-ch4emission-yeargrid-v2express
~~- [ ] noaa-cpfp-ch4-point~~
~~- [ ] noaa-cpfp-co2-point~~
~~- [ ] emit-ch4plume-v1~~

# Thumbnails

- [x] casagfed-carbonflux-monthgrid-v3
- [x] eccodarwin-co2flux-monthgrid-v5
- [x] gosat-based-ch4budget-yeargrid-v1
- [x] lpjwsl-wetlandch4-grid-v1
- [x] oco2-mip-co2budget-yeargrid-v1
- [x] oco2geos-co2-daygrid-v10r
- [x] odiac-ffco2-monthgrid-v2022
- [x] sedac-popdensity-yeargrid5yr-v4.11
- [x] tm54dvar-ch4flux-monthgrid-v1
- [x] epa-ch4emission-yeargrid-v2express
~~- [ ] noaa-cpfp-ch4-point~~
~~- [ ] noaa-cpfp-co2-point~~
~~- [ ] emit-ch4plume-v1~~

### CODE REVIEW

File: datasets/casagfed-carbonflux-monthgrid-v3.data.mdx [hunk 1]
```diff
@@ -79,6 +88,14 @@ layers:
...
       metrics:
         - mean
+    info:
+      source: NASA
+      spatialExtent: Global
+      latency: Monthly
+      unit: kg Carbon/m²/mon
+    media:
+      src: ::file ./casagfed-carbonflux-monthgrid-v3.thumbnails.npp.png
+      alt: Rendered Net Primary Production (NPP)
```
Comment (j08lue) [inline datasets/casagfed-carbonflux-monthgrid-v3.data.mdx:98]: This is how to add the thumbnails, @faustoperez. 

On second thought, I think we should not have product-specific `alt` descriptions and instead just say "Dataset rendered thumbnail" or something, to ease the config overhead.

Looks good, I think::

image

Comment (faustoperez) [reply datasets/casagfed-carbonflux-monthgrid-v3.data.mdx:98]: Thanks Jonas!

**Comment by faustoperez** [comment]
I finished adding the layer thumbnails @j08lue . I didn't update the NOAA + methane ones as they don't load on E&A. Could revisit if needed 👍

**Comment by j08lue** [comment]
> I didn't update the NOAA + methane ones as they don't load on E&A

Yes, we do not need layer thumbnails for those for now. We can add them later, if need arises.

We will need the structured data, though, please, @siddharth0248, so we can use it in new designs of the Dataset cards in the Data Catalog (https://github.com/US-GHG-Center/ghgc-architecture/issues/133).

**Comment by siddharth0248** [comment]
@faustoperez @j08lue  I have updated the mdx for all the required files and updated the checkbox list

File: datasets/casagfed-carbonflux-monthgrid-v3.data.mdx [hunk 2]
```diff
@@ -79,6 +88,14 @@ layers:
     analysis:
       metrics:
         - mean
+    info:
+      source: NASA
+      spatialExtent: Global
+      latency: Monthly
+      unit: Kilograms of carbon per square meter per month (kg Carbon/m²/mon)
```
Comment (j08lue) [inline datasets/casagfed-carbonflux-monthgrid-v3.data.mdx:95]: Units need to be in short form.

File: datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx [hunk 3]
```diff
@@ -83,6 +92,14 @@ layers:
     analysis:
       metrics:
         - mean
+    info:
+      source: NASA,NOAA
+      spatialExtent: Global
+      latency: N/A
```
Comment (j08lue) [inline datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx:98]: When not available, information should be omitted from this list, rather than set to "N/A".

**Review by j08lue — COMMENTED** [review]
Two changes we need to make to the `info` blocks:

1. Units should be abbreviated
2. Information that is not available should be omitted

File: datasets/gosat-based-ch4budget-yeargrid-v1.data.mdx [hunk 4]
```diff
@@ -34,6 +34,15 @@ taxonomy:
...
       - Model Output
+infoDescription: |
+  ::markdown
+    - Temporal Extent: 2019   
+    - Temporal Resolution: Annual   
+    - Spatial Extent: Global    
+    - Spatial Resolution: 1° x 1°    
+    - Data Units: Teragrams of methane per year (Tg CH₄/yr)
+    - Data Type: Research
+    - Data Latency: Updated yearly
```
Comment (Jeanne-le-Roux) [inline datasets/gosat-based-ch4budget-yeargrid-v1.data.mdx:45]: For ECCO-Darwin we say "Updated annually" - we should choose one way of wording this for consistency

File: datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx [hunk 5]
```diff
@@ -83,6 +92,14 @@ layers:
     analysis:
       metrics:
         - mean
+    info:
+      source: NASA,NOAA
```
Comment (Jeanne-le-Roux) [inline datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx:96]: Just checking, should there a space after the comma? or is the comma all that's needed?

**Review by Jeanne-le-Roux — APPROVED** [review]
Overall seems fine to me - I'd like to take a look once everything is in the E&A environment, we may want to adjust a few things based on how it displays. (e.g. we may want to be more concise with how we describe latency for individual data layers)

**Comment by j08lue** [comment]
> I'd like to take a look once everything is in the E&A environment

I merged the latest changes here into the preview at https://github.com/US-GHG-Center/veda-config-ghg/pull/319 - please check it out here: https://deploy-preview-319--ghg-demo.netlify.app/exploration

File: datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx (hunk 5 above)
Comment (j08lue) [reply datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx:96]: It currently looks like this when rendered:

image

I think `NASA / NOAA` might look better than the comma here. Would you agree, @Jeanne-le-Roux?

We also need to drop the "N/A" - just omit those properties, please.

Comment (Jeanne-le-Roux) [reply datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx:96]: Yes I agree, let's move forward with that

**Comment by Jeanne-le-Roux** [comment]
@j08lue When you click on an info button for a dataset layer, I think it would be more useful to see the temporal resolution on the top line as opposed to the latency. (e.g. in this example "monthly" rather than "approximately 2 years" - what do you think? Was there a justification for choosing latency to go in this position?
image

**Comment by j08lue** [comment]
> I think it would be more useful to see the temporal resolution on the top line as opposed to the latency

Yes, I think this goes for the dataset card as well as the info page. I'll get that changed in `veda-ui`.

File: datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx (hunk 5 above)
Comment (j08lue) [reply datasets/oco2-mip-co2budget-yeargrid-v1.data.mdx]: Done.

**Comment by j08lue** [comment]
I made the changes we discussed, @Jeanne-le-Roux and @siddharth0248:

1. Make the list of short info `source`, `spatialExtent`, `temporalResolution`, `unit`
2. Replace NASA,NOAA with NASA / NOAA

Can you give it one more look in this preview, please, @Jeanne-le-Roux, and confirm that it is ready to go out (when we launch E&A)?

https://deploy-preview-319--ghg-demo.netlify.app/

**Comment by Jeanne-le-Roux** [comment]
@j08lue I think this looks good. One more thing - right now the unique layer descriptions are only displayed when you are presented with layers to select, but when you click on the info button for a layer in the explore environment that short layer description doesn't seem to be there. This is something that we need to make sure is visible somewhere. 
layer_descr
layer_info

**Comment by j08lue** [comment]
Thanks, @Jeanne-le-Roux, I'll merge this PR then, to avoid more merge conflicts. Since the changes are backwards-compatible, there should be no issue in merging them already now, before the release of E&A.

I ticketed your suggestion for adding the layer description to the info modal - should be a quick fix.
- https://github.com/NASA-IMPACT/veda-ui/issues/912

**Pull request merged**